### PR TITLE
ec point format TLS extension

### PR DESCRIPTION
**Pull request by moisesguimaraes**
The ec_point_format TLS extension **MUST** be sent along with the supported_groups (supported curves) only when ECC suites are included in the client_hello message. It **MUST** also be sent back by the server if an ECC suite was chosen.

### CODE REVIEW

File: src/tls.c [hunk 1]
```diff
@@ -2948,6 +3026,24 @@ static word16 TLSX_EllipticCurve_WriteR(EllipticCurve* curve, byte* output)
     return OPAQUE16_LEN + offset;
 }
 
+#endif
+
+static word16 TLSX_PointFormat_WriteR(PointFormat* point, byte* output);
+static word16 TLSX_PointFormat_WriteR(PointFormat* point, byte* output)
```
Comment (dgarske) [inline src/tls.c]: Will you refactor the `TLSX_PointFormat_WriteR` and  `TLSX_EllipticCurve_WriteR` functions to not use recursive calls (use a while loop instead). Also you can remove the duplicate static function declaration for both; those are not required.

File: examples/client/client.c [hunk 2]
```diff
@@ -1631,18 +1631,6 @@ THREAD_RETURN WOLFSSL_THREAD client_test(void* args)
             err_sys("DisableExtendedMasterSecret failed");
         }
 #endif
-#if defined(HAVE_CURVE25519) && defined(HAVE_SUPPORTED_CURVES)
```
Comment (dgarske) [inline examples/client/client.c]: Removing this code breaks the `./examples/client/client -t` use case to force use of X25519. Please repair this and make sure there is still a way to force X25519.

**Comment by moisesguimaraes** [comment]
@dgarske : I ran both **wolfssl client** x **openssl server** and **openssl client x wolfssl server**. Wireshark was used to monitor the traffic and verify that the extension behavior was according to the RFC's specification.

I used openssl 1.0.2l for this test.

**Comment by dgarske** [comment]
Thanks Moises. Can you resolve the conflict with src/tls.c. Otherwise should be ready to go. Thanks for the info on interop testing.

**Comment by dgarske** [comment]
@toddouska : FYI: This PR is ready for final review. I don't need it for the async release.

**Pull request merged**